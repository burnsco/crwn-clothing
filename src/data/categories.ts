const productCategories = [
  { id: "burger-category", title: "Burgers" },
  { id: "burger-category", title: "Chicken" },
  { id: "burger-category", title: "Pizza" },
  { id: "burger-category", title: "Sandwiches" },
  { id: "burger-category", title: "Sushi" }
]

export default productCategories
