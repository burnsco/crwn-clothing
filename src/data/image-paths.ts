const imageData = {
  burgers: [
    { id: 18, name: "Salmon Sashimi", price: 8 },
    { id: 19, price: 5, name: "California Roll" },
    { name: "Cooked Ramen", price: 7, id: 20 },
    { price: 7, name: "Avocado Roll", id: 21 },
    { name: "Red Dragon Roll", price: 18, id: 22 }
  ],
  sushi: [
    { id: 18, name: "Salmon Sashimi", price: 8 },
    { id: 19, price: 5, name: "California Roll" },
    { name: "Cooked Ramen", price: 7, id: 20 },
    { price: 7, name: "Avocado Roll", id: 21 },
    { name: "Red Dragon Roll", price: 18, id: 22 }
  ],
  pizza: [
    { id: 18, name: "Salmon Sashimi", price: 8 },
    { id: 19, price: 5, name: "California Roll" },
    { name: "Cooked Ramen", price: 7, id: 20 },
    { price: 7, name: "Avocado Roll", id: 21 },
    { name: "Red Dragon Roll", price: 18, id: 22 }
  ],
  sandwiches: [
    { id: 18, name: "Salmon Sashimi", price: 8 },
    { id: 19, price: 5, name: "California Roll" },
    { name: "Cooked Ramen", price: 7, id: 20 },
    { price: 7, name: "Avocado Roll", id: 21 },
    { name: "Red Dragon Roll", price: 18, id: 22 }
  ],
  chicken: [
    { id: 18, name: "Salmon Sashimi", price: 8 },
    { id: 19, price: 5, name: "California Roll" },
    { name: "Cooked Ramen", price: 7, id: 20 },
    { price: 7, name: "Avocado Roll", id: 21 },
    { name: "Red Dragon Roll", price: 18, id: 22 }
  ]
}

export default imageData
