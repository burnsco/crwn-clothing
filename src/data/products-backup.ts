import firebaseAdmin from "@config/firebaseAdmin"
import { NextApiRequest, NextApiResponse } from "next"

const getAllProducts = async (req: NextApiRequest, res: NextApiResponse) => {
  try {
    const db = firebaseAdmin.firestore()

    const collections = db.collection("collections")
    const burgerDoc = await collections.doc("Burgers").get()
    const chickenDoc = await collections.doc("Chicken").get()
    const pizzaDoc = await collections.doc("Pizza").get()
    const sandwichDoc = await collections.doc("Sandwiches").get()
    const sushiDoc = await collections.doc("Sushi").get()

    const products = {
      Burgers: burgerDoc.data(),
      Chicken: chickenDoc.data(),
      Pizza: pizzaDoc.data(),
      Sandwiches: sandwichDoc.data(),
      Sushi: sushiDoc.data()
    }

    res.status(200).json(products)
  } catch (err) {
    res.status(404).json({ message: "Products not found!" })
  }
}

export default getAllProducts
