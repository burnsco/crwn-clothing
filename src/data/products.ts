const productData = {
  Burgers: {
    id: 1,
    routeName: "burgers",
    items: [
      { price: 14, id: 2, name: "Burger and Coleslaw" },
      { name: "Tomato and Cheese", price: 16, id: 3 },
      { id: 4, name: "Deluxe, Bacon and Cheese", price: 18 },
      { price: 25, name: "Super Double Burger Surprise", id: 5 },
      { name: "Yum Yum Burger", price: 14, id: 6 },
      { id: 7, name: "Burger King Clone", price: 14 },
      { name: "Basic Burger", price: 9, id: 8 },
      { name: "Cheezy", price: 17, id: 9 }
    ],
    title: "Burgers"
  },
  Chicken: {
    id: 4,
    routeName: "chicken",
    title: "Chicken",
    items: [
      { name: "Chicken and Rice", price: 25, id: 23 },
      { price: 20, id: 24, name: "Fried Chicken" },
      { id: 25, name: "Chicken on a Plate", price: 18 },
      { price: 20, name: "BBQ and Sauce", id: 26 },
      { id: 27, name: "Chicken `N Shrimp", price: 45 },
      { price: 15, name: "Chicken Leg and Veggies", id: 28 },
      { id: 29, name: "Fried Chicken Sandwich", price: 16 }
    ]
  },
  Pizza: {
    routeName: "pizza",
    items: [
      { id: 10, name: "Veggie", price: 17 },
      { name: "Baked Italian", id: 11, price: 17 },
      { name: "Tomato Classic", id: 12, price: 15 },
      { price: 12, id: 13, name: "Pepperoni" },
      { id: 14, price: 3, name: "Single Slice" },
      { price: 16, name: "XX Special Pepperoni", id: 15 },
      { id: 16, price: 17, name: "Italian Super Pepperoni" },
      { name: "Cheese", id: 17, price: 14 }
    ],
    title: "Pizza",
    id: 2
  },
  Sandwiches: {
    items: [
      { name: "White Cream on White", price: 12, id: 30 },
      { price: 9, id: 31, name: "Whole Wheat and Turkey" },
      { id: 32, price: 10, name: "Vegetale-Meat" },
      { price: 8, id: 33, name: "Vegetable" },
      { name: "Tomato and Special Sauce Bagel", id: 34, price: 12 },
      { name: "Avacado and Egg Salad", id: 35, price: 16 }
    ],
    routeName: "sandwiches",
    id: 5,
    title: "Sandwiches"
  },
  Sushi: {
    id: 3,
    title: "Sushi",
    items: [
      { id: 18, price: 8, name: "Salmon Sashimi" },
      { id: 19, price: 5, name: "California Roll" },
      { price: 7, id: 20, name: "Cooked Ramen" },
      { price: 7, name: "Avocado Roll", id: 21 },
      { name: "Red Dragon Roll", id: 22, price: 18 }
    ],
    routeName: "sushi"
  }
}

export default productData
