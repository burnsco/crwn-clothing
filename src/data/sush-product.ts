const sushiData = {
  data: {
    id: 3,
    items: [{ id: 18, name: "Salmon Sashimi", price: 8 }],
    routeName: "sushi",
    title: "Sushi"
  }
}

export default sushiData
