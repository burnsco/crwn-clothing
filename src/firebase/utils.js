import firebase from 'firebase/app'
import 'firebase/firestore'
import 'firebase/auth'

const config = {
  apiKey: 'AIzaSyC0hg2LDxJ82yZ6YoFm9ro4AD5M-YiAt0A',
  authDomain: 'crwn-db-52861.firebaseapp.com',
  databaseURL: 'https://crwn-db-52861.firebaseio.com',
  projectId: 'crwn-db-52861',
  storageBucket: 'crwn-db-52861.appspot.com',
  messagingSenderId: '767664389138',
  appId: '1:767664389138:web:8f6cb472685c0f0e59accb',
  measurementId: 'G-C3BBRHYK9T'
}

firebase.initializeApp(config)

export const auth = firebase.auth()
export const firestore = firebase.firestore()

const provider = new firebase.auth.GoogleAuthProvider()
provider.setCustomParameters({ prompt: 'select_account' })
export const signInWithGoogle = () => auth.signInWithPopup(provider)

export default firebase
