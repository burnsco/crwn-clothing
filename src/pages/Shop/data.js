const SHOP_DATA = {
  burgers: {
    id: 1,
    title: 'Burgers',
    routeName: 'burgers',
    items: [
      {
        id: 1,
        name: 'Brown Brim',
        imageUrl:
          'https://i.ibb.co/XS6jtb6/tomato-burger-and-fried-fries-1600727.jpg',
        price: 25
      },
      {
        id: 2,
        name: 'Blue Beanie',
        imageUrl:
          'https://i.ibb.co/CVJR1Hv/meat-burger-with-coleslaw-on-side-and-brown-handled-fork-156114.jpg',
        price: 18
      },
      {
        id: 3,
        name: 'Brown Cowboy',
        imageUrl:
          'https://i.ibb.co/553X9wM/meat-tomato-and-cheese-burger-161519.jpg',
        price: 35
      },
      {
        id: 4,
        name: 'Grey Brim',
        imageUrl:
          'https://i.ibb.co/TgGC8SY/hamburger-with-egg-and-vegetable-1251198.jpg',
        price: 25
      },
      {
        id: 5,
        name: 'Green Beanie',
        imageUrl:
          'https://i.ibb.co/0Kgs5dH/hamburger-and-potato-fries-1600743.jpg',
        price: 18
      },
      {
        id: 6,
        name: 'Palm Tree Cap',
        imageUrl: 'https://i.ibb.co/ys9PZY4/hamburger-161674.jpg',
        price: 14
      },
      {
        id: 7,
        name: 'Red Beanie',
        imageUrl: 'https://i.ibb.co/r2Zd0MV/cooked-foods-750073.jpg',
        price: 18
      },
      {
        id: 8,
        name: 'Wolf Cap',
        imageUrl:
          'https://i.ibb.co/HKWhkNk/bread-burger-cheese-delicious-580612.jpg',
        price: 14
      },
      {
        id: 9,
        name: 'Blue Snapback',
        imageUrl: 'https://i.ibb.co/0Xc9g01/cheese-burger-161675.jpg',
        price: 16
      }
    ]
  },
  chicken: {
    id: 4,
    title: 'Chicken',
    routeName: 'chicken',
    items: [
      {
        id: 23,
        name: 'Blue Tanktop',
        imageUrl:
          'https://i.ibb.co/hHvxVSs/plate-of-rice-and-cooked-meat-1624487.jpg',
        price: 25
      },
      {
        id: 24,
        name: 'Floral Blouse',
        imageUrl:
          'https://i.ibb.co/G5nTVCr/fried-meat-on-top-of-white-plate-1059943.jpg',
        price: 20
      },
      {
        id: 25,
        name: 'Floral Dress',
        imageUrl: 'https://i.ibb.co/v3FpKXF/food-in-a-plate-1860204.jpg',
        price: 80
      },
      {
        id: 26,
        name: 'Red Dots Dress',
        imageUrl:
          'https://i.ibb.co/R9vhtTF/barbecue-on-grill-with-sauce-platter-929192.jpg',
        price: 80
      },
      {
        id: 27,
        name: 'Striped Sweater',
        imageUrl: 'https://i.ibb.co/KmSkMbH/striped-sweater.png',
        price: 45
      },
      {
        id: 28,
        name: 'Yellow Track Suit',
        imageUrl:
          'https://i.ibb.co/nLgpsWP/cooked-leg-chicken-with-green-vegetable-leaf-60616.jpg',
        price: 135
      },
      {
        id: 29,
        name: 'White Blouse',
        imageUrl:
          'https://i.ibb.co/1R4Pt9t/american-beef-big-black-background-236486.jpg',
        price: 20
      }
    ]
  },
  sushi: {
    id: 3,
    title: 'Sushi',
    routeName: 'sushi',
    items: [
      {
        id: 18,
        name: 'Black Jean Shearling',
        imageUrl: 'https://i.ibb.co/ThkNw4b/sushi-2098143.jpg',
        price: 125
      },
      {
        id: 19,
        name: 'Blue Jean Jacket',
        imageUrl:
          'https://i.ibb.co/vJK434s/delicious-dish-epicure-fish-359992.jpg',
        price: 90
      },
      {
        id: 20,
        name: 'Grey Jean Jacket',
        imageUrl: 'https://i.ibb.co/cCm9RnH/cooked-ramen-1907244.jpg',
        price: 90
      },
      {
        id: 21,
        name: 'Brown Shearling',
        imageUrl:
          'https://i.ibb.co/gWGwcg5/avocado-ceramic-plate-chopsticks-culture-343870.jpg',
        price: 16
      },
      {
        id: 22,
        name: 'Tan Trench',
        imageUrl:
          'https://i.ibb.co/rGJSVDF/closed-up-photography-of-sushi-roll-1199985.jpg',
        price: 18
      }
    ]
  },
  pizza: {
    id: 2,
    title: 'Pizza',
    routeName: 'pizza',
    items: [
      {
        id: 10,
        name: 'Adidas NMD',
        imageUrl:
          'https://i.ibb.co/pnDzsh0/vegetables-italian-pizza-restaurant-2232.jpg',
        price: 220
      },
      {
        id: 11,
        name: 'Adidas Yeezy',
        imageUrl:
          'https://i.ibb.co/rxMW6WV/top-view-photo-of-baked-pizza-2147491.jpg',
        price: 280
      },
      {
        id: 12,
        name: 'Black Converse',
        imageUrl: 'https://i.ibb.co/RcXsYcr/tomato-pizza-208537.jpg',
        price: 110
      },
      {
        id: 13,
        name: 'Nike White AirForce',
        imageUrl:
          'https://i.ibb.co/bPJpbb2/sliced-pepperoni-pizza-on-white-ceramic-plate-708587.jpg',
        price: 160
      },
      {
        id: 14,
        name: 'Nike Red High Tops',
        imageUrl: 'https://i.ibb.co/9ZWFZqB/pizza-slice-842519.jpg',
        price: 160
      },
      {
        id: 15,
        name: 'Nike Brown High Tops',
        imageUrl: 'https://i.ibb.co/rcz8vhc/pepperoni-pizza-803290.jpg',
        price: 160
      },
      {
        id: 16,
        name: 'Air Jordan Limited',
        imageUrl: 'https://i.ibb.co/89010rr/baked-pepperoni-pizza-774487.jpg',
        price: 190
      },
      {
        id: 17,
        name: 'Timberlands',
        imageUrl:
          'https://i.ibb.co/6Np31JL/cheese-close-up-cooking-crispy-263041.jpg',
        price: 200
      }
    ]
  },
  sandwiches: {
    id: 5,
    title: 'Sandwiches',
    routeName: 'sandwiches',
    items: [
      {
        id: 30,
        name: 'White Cream on White',
        imageUrl:
          'https://i.ibb.co/pvvCT40/white-cream-on-white-bowl-1633525.jpg',
        price: 12
      },
      {
        id: 31,
        name: 'Whole Wheat and Turkey',
        imageUrl: 'https://i.ibb.co/QnygRz2/wheat-bread-sandwich-1209029.jpg',
        price: 9
      },
      {
        id: 32,
        name: 'Vegetale-Meat',
        imageUrl:
          'https://i.ibb.co/tY4qvsT/vegetale-and-meat-sandwhich-133578.jpg',
        price: 10
      },
      {
        id: 33,
        name: 'Vegetable',
        imageUrl: 'https://i.ibb.co/Jr3Q5SV/vegetable-sandwich-1988624.jpg',
        price: 8
      },
      {
        id: 34,
        name: 'Tomato and Special Sauce Bagel',
        imageUrl: 'https://i.ibb.co/M579tn0/sliced-sandwich-1630309.jpg',
        price: 12
      },
      {
        id: 35,
        name: 'Avacado and Egg Salad',
        imageUrl:
          'https://i.ibb.co/72xpFhK/bread-breakfast-cheese-food-236813.jpg',
        price: 16
      }
    ]
  }
}

export default SHOP_DATA
