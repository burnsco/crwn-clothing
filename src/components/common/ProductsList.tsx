import ProductCard from "@components/common/ProductCard"
import "firebase/firestore"

export default function ProductsList(props: any) {
  const { data } = props
  const { items } = data
  const { routeName } = props.data

  return (
    <article className="container">
      <section className="grid grid-cols-1 md:grid-cols-3 sm:grid-cols-2 gap-10">
        {items.map((item: any, i: number) => (
          <ProductCard
            key={`product-${item}-${i}`}
            item={item}
            routeName={routeName}
          />
        ))}
      </section>
    </article>
  )
}
