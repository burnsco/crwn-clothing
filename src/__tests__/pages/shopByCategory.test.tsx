import ProductsList from "@components/common/ProductsList"
import "@testing-library/jest-dom"
import data from "../../data/sushi-products"
import { render } from "../../utils/testUtils"

const useRouter = jest.spyOn(require("next/router"), "useRouter")

describe("Category Products Page", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<ProductsList {...data} />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it("Renders all the products (ie. sushi)", async () => {
    useRouter.mockImplementation(() => ({
      route: "/shop/[category]",
      pathname: "/shop/[category]",
      query: { category: "sushi" },
      asPath: "/shop/sushi"
    }))
    const { getByText } = render(<ProductsList {...data} />)

    expect(getByText(/salmon sashimi/i)).toBeInTheDocument()
    expect(getByText(/california roll/i)).toBeInTheDocument()
    expect(getByText(/cooked ramen/i)).toBeInTheDocument()
    expect(getByText(/avocado roll/i)).toBeInTheDocument()
    expect(getByText(/red dragon roll/i)).toBeInTheDocument()
  })
})
