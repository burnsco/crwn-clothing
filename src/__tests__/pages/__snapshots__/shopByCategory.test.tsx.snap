// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Products Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-1au2bwr"
  >
    <div
      class="simple-grid-shop css-1vfkk4i"
    />
  </div>
</DocumentFragment>
`;
